ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx

```diff
@@ -0,0 +1,152 @@
+/*---------------------------------------------------------------------------------------------
```

Comment:
I must admit having hovered and focused rule items both highlighted at the same time feels off. I don't have a good solution for this, except maybe not highlighting focused node after the mouse is moved onto another node.

Reply:
I think this problem and the bug that I found above can be both addressed by having a single piece of state which indicates the active node instead of hosting this state inside each of them.